src/eduid_webapp/email/views.py

```diff
@@ -213,9 +218,13 @@ def verify_link(user):
     email = request.args.get('email')
     if code and email:
         current_app.logger.debug('Trying to save email address {} as verified for user {}'.format(email, proofing_user))
-
         db = current_app.proofing_statedb
-        state = db.get_state_by_eppn_and_email(proofing_user.eppn, email)
+        state = db.get_state_by_eppn_and_email(proofing_user.eppn, email, raise_on_missing=False)
+
+        if state is None:
+            current_app.logger.info("Verification code unknown: {}. Sending new code".format(code))
```

Comment:
It's not the code that is unknown, there is no state for this eppn+email - right?

Reply:
Fixed